### DISCUSSION TITLE
Homebrew gets confused with newer Xcode

### DISCUSSION THREAD

**Discussion by exadeci** [Homebrew/discussions#3102] [Everyday usage]
Hello,

I'm on Monterey 12.2.1 (21D62) with Xcode 13.3 and matching 13.3 command-line tools.

Brew refuses to install a package telling me that:
`Error: Your Command Line Tools are too outdated.`
`You should download the Command Line Tools for Xcode 13.2.1.`

And it suggests me to do a `softwareupdate --all --install --force` which is a full system upgrade with a restart to Monterey 12.3 (**with no warnings!**).

I don't get why Brew is complaining about having a newer version of the command line tools? Do you need to add them manually?

```
λ brew --version                                                                                                                                            
Homebrew 3.4.3
Homebrew/homebrew-core (git revision 1d406953b03; last commit 2022-03-22)
Homebrew/homebrew-cask (git revision ddcef90bc0; last commit 2022-03-22)
```

**Comment by Bo98** [comment]
What version is `brew config` reporting?

**Reply by grepe** [reply]
Same issue here on, macOS 12.3 (21E230)

Output of `brew config`
```
~ % brew config
HOMEBREW_VERSION: 3.4.3
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 600d84e56eda931c3dd3b6a9ee61233d9bf2b054
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 49a025a7429cc4b216b60684b08d7b94a550a96f
Core tap last commit: 44 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: :0
HOMEBREW_EDITOR: /usr/bin/emacs
HOMEBREW_MAKE_JOBS: 6
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: hexa-core 64-bit kabylake
Clang: 13.1.6 build 1316
Git: 2.35.1 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3-x86_64
CLT: 12.0.0.32.29
Xcode: 13.3
```

and
````
% xcode-select -v       
xcode-select version 2395.
````

**Reply by SMillerDev** [reply]
What does `xcode-select -p` say?

**Reply by Bo98** [reply]
> `12.0.0.32.29`

This is CLT 12.4. How did you install CLT 13.3?

**Comment by exadeci** [comment]
@Bo98 

I have since installed the older 13.2.1 so that I could install the package so it might be different from what it was but here it's:

```
HOMEBREW_VERSION: 3.4.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 600d84e56eda931c3dd3b6a9ee61233d9bf2b054
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 93012c1045039fb00e0862ee5e62d99cd8311eb6
Core tap last commit: 61 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: code -w
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 13.1.6 build 1316
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.2.1-x86_64
CLT: 13.2.0.0.1.1637719136
Xcode: 13.3
```

**Comment by grepe** [comment]
> On 22 Mar 2022, at 10:43, Sean Molenaar ***@***.***> wrote:
> 
> ﻿
> What does xcode-select -p say?
> 

/Applications/Xcode.app/Contents/Developer

(note: I also had to revert to 13.2 to finish the install. I assume that the path is still the same.)

Thanks,
Peter.

**Reply by SMillerDev** [reply]
If you set the path to the system CLT (`/Library/Developers`) is brew still confused? Additionally, what does `brew doctor` say?

**Reply by jaredcat** [reply]
`sudo xcode-select --switch /Library/Developer/CommandLineTools` fixed it for me

**Comment by lgruen** [comment]
Manually installing the [Command Line Tools for Xcode 13.3](https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_13.3/Command_Line_Tools_for_Xcode_13.3.dmg) worked for me. It didn't show up in `softwareupdate --all --install`.
